### ISSUE TITLE
Tuya Climate does not handle C/F gracefully

### ISSUE THREAD

**Issue by user897943** [esphome/issues#3352]
### The problem

I set up a Tuya Climate component for a portable air conditioner. The AC supports temperature setpoints from 16C to 31C in 1C increments. The AC has a button to change unit between C and F, but even when using F it still uses C to set the MCU parameter, I believe what the MCU does is just to round to the nearest C temperature when selecting an unsupported Fahrenheit one, and report that it successfully set the command.. So for example:
I set 77F on HA, that would be 25C, so the unit accepts the value
I set 78C on HA, this is 25.5556C which is unsupported, so the unit does not accept the value and the frontend does not show it.
If I use the unit's remote control and set to 78, the unit does accept the value but I can see on the logs that it simply rounds to the next integer!
So in a nutshell, please fix the code to always round to the next integer if the appliance only supports C on the board (even if it has a button to alternate scales), and then let the F value remain as-is for HA, so that if I set to 78, I get 78 on the frontend but the AC is actually setting itself to 26 (rounded up from 25.5556).

### Which version of ESPHome has the issue?

2022.6.0-dev

### What type of installation are you using?

Docker

### Which version of Home Assistant has the issue?

2022.6.2

### What platform are you using?

ESP32

### Board

widora-air (ESP32 16MB flash)

### Component causing the issue

tuya/climate

**Comment by user897943** [comment]
Still happening
